profile/src/v100/app_preferences/gateways/gateway.rs

```diff
@@ -0,0 +1,277 @@
+use crate::prelude::*;
+use std::ops::Deref;
+
+/// A gateway to some Radix Network, which is a high level REST API which clients (wallets) can
+/// consume in order to query asset balances and submit transactions.
+#[derive(
+    Serialize,
+    Deserialize,
+    Clone,
+    PartialEq,
+    Eq,
+    Ord,
+    PartialOrd,
+    Hash,
+    derive_more::Display,
+    derive_more::Debug,
+    uniffi::Record,
+)]
+#[display("{}", self.url.to_string())]
+#[debug("{}: {}", self.network.display_description, self.url.to_string())]
+pub struct Gateway {
+    /// The Radix network the API is a Gateway to.
+    pub network: RadixNetwork,
```

Comment:
The disambiguation is `RadixNetwork` is very close to `radix_engine_common::network::NetworkDefinition`, perhaps we should call `RadixNetwork` -> `NetworkDefinition` instead...?
And perhaps we should rename the `Network` (element of `profile.networks`) -> `ProfileNetwork`?